src/polydb.jl

```diff
@@ -0,0 +1,284 @@
+
+function Base.iterate(coll::Collection, state::Mongoc.Cursor)
+   return _evaluateIteration(iterate(coll.mcol, state))
+end
+
+# functions for `BSON` iteration
+function Base.iterate(cursor::BSONCursor)
+   return iterate(cursor.mcursor)
+end
+
+function Base.iterate(cursor::BSONCursor, state::Nothing)
+   return iterate(cursor.mcursor, state)
+end
+
+function Base.collect(cursor::BSONCursor)
+   return collect(cursor.mcursor)
+end
+
+function Base.iterate(coll::BSONCollection)
+   return iterate(coll.mcol)
+end
+
+function Base.iterate(coll::BSONCollection, state::Mongoc.Cursor)
+   return iterate(coll.mcol, state)
+end
+
+#Info
+
+# prints information about a specific Collection
+# also used for the info(::Database) function
+function info(coll::Collection)
+   db = coll.mcol.database
+   coll_c = db[string("_collectionInfo.", coll.mcol.name)]
+   info = iterate(coll_c)[1]
+   _print_collection(info)
+end
+
+# returns an array containing the names of all collections in the Polydb
+function _get_collection_names(db::Database)
+   opts = Mongoc.BSON("authorizedCollections" => true, "nameOnly" => true)
+   return Mongoc.get_collection_names(db.mdb;options=opts)
+end
+
+# for the set of names obtained by the _get_collection_names(::Database) function
+# returns two arrays containing the names of the meta data collections
+# first one for sections, second one for collections
+function _get_meta_names(names::Array{String, 1})
+   n = length(names)
+   sec_bool = Array{Bool, 1}(undef, n)
+   coll_bool = Array{Bool, 1}(undef, n)
+   n_secs = 0
+   n_colls = 0
+   i = 1
+   for name in names
+      if SubString(name, 1, 2) == "_s"
+         sec_bool[i] = true
+         coll_bool[i] = false
+         n_secs += 1
+      elseif SubString(name, 1, 2) == "_c"
+         sec_bool[i] = false
+         coll_bool[i] = true
+         n_colls += 1
+      else
+         sec_bool[i] = false
+         coll_bool[i] = false
+      end
+      i += 1
+   end
+   secs = Array{String,1}(undef, n_secs)
+   colls = Array{String,1}(undef, n_colls)
+   i_s = 1
+   i_c = 1
+   for j = 1:n
+      if sec_bool[j]
+         secs[i_s] = names[j]
+         i_s += 1
+      elseif coll_bool[j]
+         colls[i_c] = names[j]
+         i_c += 1
+      end
+   end
+   return secs, colls
+end
+
+# functions helping printing metadata for sections or collections
+function _print_contact(s::String)
+   println(s)
+end
+
+function _print_contact(a::Array)
+   for dict in a
+      str = dict["name"]
+      for key in ["email", "www", "affiliation"]
+         if haskey(dict, key) && dict[key] != ""
+            str = string(str, ", ", dict[key])
+         end
+      end
+      println(string("\t\t", str))
+   end
+end
+
+# prints information about a specific section and
+# continues to print information about its content
+function _print_section(db::Database, info::Mongoc.BSON, sections::Array{String,1}, collections::Array{String,1})
+   @info string("SECTION: ", join(info["section"], "."))
```

Comment:
note that `@info` goes to `Logging.get_logger()`, while `println` goes to `stdout`. If I capture logs to file I won't see anything except `@info` line

again, it'd be better to separate getting 'string' representation from printing/logging